machineconfiguration/v1alpha1/custom.pinnedimageset.testsuite.yaml

```diff
@@ -0,0 +1,87 @@
+crd: 0000_80_machine-config_01_pinnedimagesets-CustomNoUpgrade.crd.yaml
+tests:
+  onCreate:
+  - name: Should be able to create a minimal PinnedImageSet
+    initial: |
+      apiVersion: machineconfiguration.openshift.io/v1alpha1
+      kind: PinnedImageSet
+      metadata:
+        name: test-pinnedimageset
+      spec:
+        pinnedImages:
+          - name: "example.io/my-project/image-v1.0_23@sha256:2c3ea52ac3a41c6d58e85977c3149413e3fa4b70eb2397426456863adbf43306"
+          - name: "example.com:5000/project/image@sha256:abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+          - name: "example.com/image@sha256:abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+          - name: "my-registry.local/project/my-image@sha256:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"
+    expected: |
+      apiVersion: machineconfiguration.openshift.io/v1alpha1
+      kind: PinnedImageSet
+      metadata:
+        name: test-pinnedimageset
+      spec:
+        pinnedImages:
+          - name: "example.io/my-project/image-v1.0_23@sha256:2c3ea52ac3a41c6d58e85977c3149413e3fa4b70eb2397426456863adbf43306"
+          - name: "example.com:5000/project/image@sha256:abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+          - name: "example.com/image@sha256:abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890"
+          - name: "my-registry.local/project/my-image@sha256:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"
+  - name: Invalid image referenced by tag
+    initial: |
+      apiVersion: machineconfiguration.openshift.io/v1alpha1
+      kind: PinnedImageSet
+      metadata:
+        name: test-pinnedimageset
+      spec:
+        pinnedImages:
+          - name: "example.io/my-project/image-v1.0_23:latest"
+    expectedError: "Invalid value: \"string\": The OCI Image reference must be in the format host[:port][/namespace]/name@sha256:<digest> with a valid SHA256 digest"
+  - name: Invalid empty image
+    initial: |
+      apiVersion: machineconfiguration.openshift.io/v1alpha1
+      kind: PinnedImageSet
+      metadata:
+        name: test-pinnedimageset
+      spec:
+        pinnedImages:
+          - name: ""
+    expectedError: "Invalid value: \"string\": The OCI Image reference must be in the format host[:port][/namespace]/name@sha256:<digest> with a valid SHA256 digest"
+  - name: Should reject a malformed SHA256 digest
+    initial: |
+      apiVersion: machineconfiguration.openshift.io/v1alpha1
+      kind: PinnedImageSet
+      metadata:
+        name: test-pinnedimageset-bad-digest
+      spec:
+        pinnedImages:
+          - name: "example.io/bad-digest/image@sha256:12345"
+    expectedError: "Invalid value: \"string\": The OCI Image reference must be in the format host[:port][/namespace]/name@sha256:<digest> with a valid SHA256 digest"
+  - name: Should accept multiple images with different registries and namespaces
```

Comment:
Do we have a reject identical images test?

Reply:
added